lib/ignored-paths.js

```diff
@@ -201,6 +201,12 @@ class IgnoredPaths {
 
         const ig = ignore().add(DEFAULT_IGNORE_DIRS);
 
+        if (this.options.dotfiles !== true) {
+
+            // Ignore hidden folders.  (This cannot be ".*", or else it's not possible to unignore hidden files)
+            ig.add([".*/*", "!../"]);
```

Comment:
Yeah, I added that pattern a while ago because the patterns in `.eslintignore` currently are applied against the CWD, not the file's directory (until https://github.com/eslint/eslint/pull/7678 is merged). I think we could consider removing that pattern in that PR, since you're right, from then on, the patterns will be relative to the ignore file's location, and the ignore file must be at the project root (unless we make changes for https://github.com/eslint/eslint/issues/8090).

It might be interesting to do a search for `.eslintignore` files on github which contain `../`.  Seems like that should be possible, but I'm not exactly sure how.